vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/GridColumnOrderHelper.java

```diff
@@ -174,8 +174,8 @@ private void reorderColumnsAndConsumeIDs(Component column,
 
         // holds the current immediate child columns.
         final Set<AbstractColumn<?>> childColumns = new HashSet<>();
-        column.getChildren().filter(c -> !(c instanceof GridSelectionColumn))
-                .forEach(it -> childColumns.add((AbstractColumn) it));
+        column.getChildren().filter(c -> c instanceof AbstractColumn<?>)
+                .forEach(it -> childColumns.add((AbstractColumn<?>) it));
```

Comment:
The `!(c instanceof GridSelectionColumn)` check is obsolete as `GridSelectionColumn` doesn't extend from `AbstractColumn`.